lib/src/source/hosted.dart

```diff
@@ -186,7 +193,10 @@ class BoundHostedSource extends CachedSource {
           expectedName: ref.name, location: url);
       var id = source.idFor(ref.name, pubspec.version,
           url: _serverFor(ref.description));
-      return MapEntry(id, pubspec);
+      final archiveUrlValue = map['archive_url'];
+      final archiveUrl =
+          archiveUrlValue is String ? Uri.tryParse(archiveUrlValue) : null;
```

Comment:
What do we do if the value is `null`?

Reply:
Well, then we cannot download the file.
We could make a fallback to the hardcoded value.